builtin/providers/google/resource_google_project.go

```diff
@@ -210,6 +231,18 @@ func resourceGoogleProjectRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("policy_etag", pol.Etag)
 	d.Set("policy_data", string(polBytes))
 
+	// Read the billing account
+	ba, err := config.clientBilling.Projects.GetBillingInfo("projects/" + pid).Do()
```

Comment:
Hmm, I'm not seeing this. I've created a few projects (and just added tests that do the same) that don't have billing accounts but make it through read. GetBillingInfo should return an empty billing account if none is set. Possible to see your config?